code/game/objects/structures/extinguisher.dm

```diff
@@ -142,13 +146,56 @@
 			new /obj/item/wallframe/extinguisher_cabinet(loc)
 		else
 			new /obj/item/stack/sheet/metal (loc, 2)
-		if(stored_extinguisher)
-			stored_extinguisher.forceMove(loc)
-			stored_extinguisher = null
+		if(stored_object)
+			stored_object.forceMove(loc)
+			stored_object = null
 	qdel(src)
 
 /obj/item/wallframe/extinguisher_cabinet
 	name = "extinguisher cabinet frame"
 	desc = "Used for building wall-mounted extinguisher cabinets."
 	icon_state = "extinguisher"
 	result_path = /obj/structure/extinguisher_cabinet
+
+//wall mounted medkits
+/obj/structure/extinguisher_cabinet/medkit
+	name = "medkit cabinet"
+	desc = "A small wall mounted cabinet designed to hold a first aid kit."
+	icon = 'icons/obj/wallmounts.dmi'
+	icon_state = "medkit_closed"
+	var/obj/item/storage/firstaid/regular/stored_medkit
+	//sets allowed item for the cabinet. Setting it to main type will allow subtypes.
+	alloweditem = /obj/item/storage/firstaid
+
+//mapcode
+/obj/structure/extinguisher_cabinet/medkit/Initialize(mapload, ndir, building)
+	. = ..()
+	if(building)
+		setDir(ndir)
+		pixel_x = (dir & 3)? 0 : (dir == 4 ? -27 : 27)
+		pixel_y = (dir & 3)? (dir ==1 ? -30 : 30) : 0
+		opened = TRUE
+		icon_state = "medkit_empty"
+	else
+		//set this for item spawned inside
+		stored_object = new /obj/item/storage/firstaid/regular/(src)
+
+//sprite code
+/obj/structure/extinguisher_cabinet/medkit/update_icon()
+	if(!opened)
+		icon_state = "medkit_closed"
+		return
+	if(stored_object)
+		if(istype(stored_object, /obj/item/storage/firstaid/regular)) //leave this in case someone decides to add more sprites for different kits

```

Comment:
If someone decides to add more sprites to this, they can just add this code themselves. I would remove it.

Reply:
made sprites vars so this no longer applies.